lib/rototiller/task/params/command.rb

```diff
@@ -161,15 +160,40 @@ def run
         #   we may have to convert this to a class if it gets complex
         @result = Result.new
         @result.output = ''
-        # add ';' to command string as it is a metacharacter that forces open3
-        #   to send the command to the shell. This returns the correct
-        #   exit_code and stderr, etc when the command is not found
-        Open3.popen2e(self.to_str + ";"){|stdin, stdout_err, wait_thr|
-          stdout_err.each { |line| puts line
-                            @result.output << line }
-          @result.pid    = wait_thr.pid # pid of the started process.
-          @result.exit_code = wait_thr.value.exitstatus # Process::Status object returned.
-        }
+
+        read_pipe, write_pipe = IO.pipe
+        begin
+          @result.pid = Process.spawn(self.to_str, :out => write_pipe, :err => write_pipe)
+        rescue Errno::ENOENT => e
+          $stderr.puts e
+          @result.output << e.to_s
+          @result.exit_code = 127
+          raise
```

Comment:
no.  it re-raises the same exception with the default message.  but i've handled all our result issues in the handler here.
you can see, because this is caught in the unit tests:
https://github.com/puppetlabs/rototiller/pull/73/files#diff-02ef5a5819402594008a7cb344cd6509R61